cmd/isp/src/cmd.rs

```diff
@@ -0,0 +1,218 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::protocol::*;
+use anyhow::Result;
+
+enum DataPhase<'a> {
+    NoData,
+    Send { code: ResponseCode, data: &'a Vec<u8> },
```

Comment:
Nit: Storing a reference to a `Vec` is a little weird – could you make it a `&'a [u8]` instead?